pkg/registry/snapshot/snapshot.go

```diff
@@ -38,20 +45,135 @@ func (r *REST) GroupVersionKind(containingGV schema.GroupVersion) schema.GroupVe
 }
 
 func (r *REST) Get(ctx apirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
-	fmt.Println(">>>>--- GET")
-	return r.New(), nil
+	ns, ok := apirequest.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("missing namespace")
+	}
+	if len(name) < 9 {
+		return nil, errors.New("invalid snapshot name")
+	}
+	snapshotIDs := make([]string, 0)
+	snapshotIDs = append(snapshotIDs, name[len(name)-8:])
+
+	repoName := strings.TrimSuffix(name, name[len(name)-9:])
+
+	repo, err := r.stashClient.StashV1alpha1().Repositories(ns).Get(repoName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("respective repository not found. error:" + err.Error())
+	}
+
+	snapshots := make([]api.Snapshot, 0)
+	if repo.Spec.Backend.Local != nil {
+		snapshots, err = r.getSnapshotsFromSidecar(repo, snapshotIDs)
+	} else {
+		snapshots, err = r.GetSnapshots(repo, snapshotIDs)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if len(snapshots) == 0 {
+		return nil, errors.New("no resource found")
+	}
+
+	snapshot := &api.Snapshot{}
+	snapshot = &snapshots[0]
+	return snapshot, nil
 }
 
 func (r *REST) NewList() runtime.Object {
 	return &api.SnapshotList{}
 }
 
 func (r *REST) List(ctx apirequest.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
-	fmt.Println(">>>>--- LIST")
-	return r.NewList(), nil
+
+	ns, ok := apirequest.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("missing namespace")
+	}
+	var labelSelector string
+	if options.LabelSelector != nil {
+		labelSelector = options.LabelSelector.String()
+	}
+	repos := make([]v1alpha1.Repository, 0)
+
+	// if the request is for snapshots of specific repositories then use Get method to get those repositories
+	// otherwise use labelSelector to list all repositories that match the selectors.
+	if strings.Contains(labelSelector, "repository=") {
```

Comment:
Convert to `Requirements` and test using that https://github.com/appscode/stash/blob/master/vendor/k8s.io/apimachinery/pkg/labels/selector.go#L53